tests/Functional/Schema/SchemaManagerFunctionalTestCase.php

```diff
@@ -579,6 +579,32 @@ public function testListForeignKeys(): void
         self::assertEquals('CASCADE', $fkeys[0]->getOption('onDelete'));
     }
 
+    public function testCreateForeignKeyGh5485(): void
+    {
+        $this->createTestTable('test_create_fk1');
+        $this->createTestTable('test_create_fk2');
+
+        $table = $this->schemaManager->listTableDetails('test_create_fk1');
+        $table->addForeignKeyConstraint(
+            'test_create_fk2',
+            ['foreign_key_test'],
+            ['id'],
+            [],
+            'i'
+        );
+        $foreignKey = $table->getForeignKeys()['i'];
+        $this->schemaManager->createForeignKey($foreignKey, $table);
+
+        $fkeys = $this->schemaManager->listTableForeignKeys('test_create_fk1');
+
+        self::assertEquals(1, count($fkeys), "Table 'test_create_fk1' has to have one foreign key.");
```

Comment:
You can use `assertCount()` instead.

Reply:
I think, @mvorisek has copied this snippet from another test. Let's fix this everywhere: #5535

Reply:
yes, I will update once your PR is merged

Reply:
done